### PR TITLE
Notification listen refactor

### PR DESCRIPTION
**Pull request by Flameeyes**
Minor cleanups, I don't think this is actually needed for my changes to Home Assistant at this point, but I would think it's an improvement anyway.

### CODE REVIEW

File: songpal/device.py [hunk 1]
```diff
@@ -449,38 +454,35 @@ def clear_notification_callbacks(self):
...
-
-        for serv in self.services.values():
-            tasks.append(
-                asyncio.ensure_future(
-                    serv.listen_all_notifications(handle_notification)
+        async def handle_notification(notification: ChangeNotification) -> None:
+            callbacks = self.callbacks.get(type(notification), [fallback_callback])
+            if not callbacks:
+                _LOGGER.debug("No callbacks defined for %s", notification)
+            else:
```
Comment (rytilahti) [inline songpal/device.py]: If I'm not misreading this, would it make sense to return when no callbacks are set, and avoid indent for the else-block altogether?

Comment (Flameeyes) [reply songpal/device.py]: I have no preference between the two, done :)

Comment (rytilahti) [reply songpal/device.py]: Thanks! Less nesting makes code cleaner, and clean code is good ;-)

**Review by rytilahti — APPROVED** [review]
LGTM, thanks! :+1:

**Pull request merged**